analysis/Scripts/Efficiencies/efficiencies.py

```diff
@@ -0,0 +1,191 @@
+					cuts = Imports.getDataCuts(run)
+
+				# # APPARENTLY THAT THE ONLY CUTS THAT SIMON WAS LOOKING AT ---- !!!QUESTION!!!
+				# cuts = "lcplus_P < 300000 && lcplus_OWNPV_CHI2 < 80 &&  pplus_P < 120000 && kminus_P < 115000 && piplus_P < 80000"
+
+				# #turbo = "lcplus_Hlt2CharmHadXicpToPpKmPipTurboDecision_TOS==1"
+				# turbo = "lcplus_Hlt2CharmHadD2HHHDecision_TOS == 1"
+
+				Lc_MC_tree = TChain("tuple_Lc2pKpi/DecayTree") # !!! QUESTION : NOT BETTER ISTEAD OF CHAIN; JUST GETENTRIES FROM EACH ONE BY ONE, ONCE WITHOUT CUT AND ONCE WITH?
+
+				for subjob in os.listdir(RAW_TUPLE_PATH + job):
+					Lc_MC_tree.Add(RAW_TUPLE_PATH + job + "/" + subjob + "/" + filename)
+
+				N= float(Lc_MC_tree.GetEntries()) #WHY DID SIMON USE A HARDCODED NUMBER OF ENTRIES??
+				#k = float(Lc_MC_tree.GetEntries(cuts + " && " + turbo)) SIMON VERSION
+				k = float(Lc_MC_tree.GetEntries(cuts))
+				eff = float(k/N)
+				binom_error = (1/N)*((k*(1-k/N))**(0.5))
+
+				string = "Particle: " + particle + " year: " + str(year) + pol + " efficiency for the selection: " + cuts + " is: " + str(eff) + " +/- " + str(binom_error) + "\n"
+				f_text.write(string)
+
+				selecEffDict[particle + "_" + str(year) + "_" + pol] = {'val': eff, 'err': binom_error}
+
+
+			print("\nSelection efficiency calculations are done!")
+
+			prettyEffDict = pprint.pformat(selecEffDict)
+			dictF = open("Selection_Eff_Dict.py","w")
+			dictF.write("selDict = " + str(prettyEffDict))
+			dictF.close()
+
+			f_text.close()
+
+		elif opt == "-t":
+
+			print("\nCreation of the trigger efficiency files")
+
+			trigEffDict = {}
+
+			f_text = open("Trigger_Eff_output.txt", "w+")
+
+			n = len(MC_jobs_Dict)
+			i = 0
+
+			for job in MC_jobs_Dict:
+				#FOR THE PROGRESSION BAR
+				if i < n:
+					j = (i + 1) / n
+					sys.stdout.write('\r')
+					sys.stdout.write("[%-20s] %d%%" % ('='*int(20*j), 100*j))
+					sys.stdout.flush()
+					i += 1
+				if job == "NA":
+					continue
+
+				particle = MC_jobs_Dict[job][3]
+				year = MC_jobs_Dict[job][0]
+				pol = MC_jobs_Dict[job][1]
+				subjobs = MC_jobs_Dict[job][2]
+				identifier = MC_jobs_Dict[job][4]
+
+				filename = "MC_Lc2pKpiTuple_" + identifier + ".root"
+
+				if int(year) <= 2012:
+					run = 1
+					cuts = "lcplus_L0HadronDecision_TOS == 1 && lcplus_Hlt1TrackAllL0Decision_TOS == 1"
+					turbo = "lcplus_Hlt2CharmHadD2HHHDecision_TOS==1"
+				else:
+					run = 2
+					cuts = Imports.getDataCuts(run)
+					turbo = "lcplus_Hlt1TrackMVADecision_TOS==1"
+
+				#WHAT DO I NEED FOR TRIGGER CUTS?!!
+
+				Lc_MC_tree = TChain("tuple_Lc2pKpi/DecayTree") # !!! QUESTION : NOT BETTER ISTEAD OF CHAIN; JUST GETENTRIES FROM EACH ONE BY ONE, ONCE WITHOUT CUT AND ONCE WITH?
+
+				for subjob in os.listdir(RAW_TUPLE_PATH + job):
+					Lc_MC_tree.Add(RAW_TUPLE_PATH + job + "/" + subjob + "/" + filename)
+
+				N=float(Lc_MC_tree.GetEntries( turbo + " && lcplus_L0HadronDecision_TOS==1"))
+				k = float(Lc_MC_tree.GetEntries(cuts + " && " + turbo ))
```

Comment:
Should check with @jackfrost373 if this is the right procedure

Reply:
Reference to the above change request, this is the wrong way to calculate the efficiencies, but I will fix it when I delve back into the measuring of the efficiencies. I will then do PID and fix the cuts we do here that are not needed